ptrack/encoder.py

```diff
@@ -20,27 +22,35 @@ class PtrackEncoder(object):
     """PtrackEncoder class."""
 
     @staticmethod
-    def decrypt(encoded_data):
-        """Decrypt a base64 url string into a dictionary."""
+    def decrypt(encoded_data: Union[bytes, str]) -> Tuple[Tuple[Any], Dict[str, Any]]:
+        """Return args and kwargs decrypted from base64 url string."""
+        if isinstance(encoded_data, str):
+            encoded_data = encoded_data.encode('utf8')
+
         key = pad(settings.PTRACK_SECRET).encode('utf8')
         box = nacl.secret.SecretBox(key)
 
-        encrypted = base64.urlsafe_b64decode(encoded_data.encode('utf8'))
+        encrypted = base64.urlsafe_b64decode(encoded_data)
         data = box.decrypt(encrypted)
         # json.loads expects a str, so we convert bytes to str
         data = data.decode('utf8')
         return json.loads(data)
 
     @staticmethod
-    def encrypt(*args, **kwargs):
-        """Encrypt args and kwargs into an encoded base64 url string."""
+    def encrypt(*args, **kwargs) -> bytes:
+        """Return a base64 byte string with encoded positional/keyword args."""
         key = pad(settings.PTRACK_SECRET).encode('utf8')
         box = nacl.secret.SecretBox(key)
         nonce = nacl.utils.random(nacl.secret.SecretBox.NONCE_SIZE)
 
         data = json.dumps((args, kwargs))
         # box expects bytes, so we convert here
-        data = data.encode('utf8')
-        encrypted = box.encrypt(data, nonce)
+        bytes_data = data.encode('utf8')
+        encrypted = box.encrypt(bytes_data, nonce)
         encoded_data = base64.urlsafe_b64encode(encrypted)
         return encoded_data
+
+    @staticmethod
+    def encrypt_string(*args, **kwargs) -> str:
```

Comment:
I added an `encrypt_string` method that casts the output of `encrypt` from a bytestring to a string. I think modifying `encrypt` will probably be the right solution, but I wanted to discuss first.

Reply:
I'd be all for doing that